clash-lib/src/Clash/Driver/Manifest.hs

```diff
@@ -65,7 +72,58 @@ data Manifest
   , fileNames :: [(FilePath, ByteString)]
     -- ^ Names and hashes of all the generated files for the @TopEntity@. Hashes
     -- are SHA256.
-  } deriving (Show,Read)
+  } deriving (Show,Read,Eq)
+
+instance ToJSON Manifest where
+  toJSON (Manifest{..}) =
+    Aeson.object
+      [ "version" .= ("unstable" :: Text)
```

Comment:
I assume this is manifest format version. Do we also care about clash version? I would expect if a user updates Clash then components should be rebuilt as the compiler may output something different

Reply:
Yeah, that's all included in the `hash` field. This is just there to indicate to users that they really shouldn't rely on any API defined by the file. (We might later stabilize it though.)

Reply:
Having said that, having the Clash version+git hash that generated the file would be cool even if it's just for diagnostics.

Reply:
I've used [this](https://hackage.haskell.org/package/githash) before to bake it into a constant. We could probably use some `CPP` magic to have some `clashVersion` constant set to either the git tree + hash or the actual version number

Reply:
Sure, I'm going to leave this for a separate PR though as it's not really something we include right now. (And I think it might affect some other things we need to discuss too.)